apm-agent-plugins/apm-es-restclient-plugin/apm-es-restclient-plugin-5_6/src/main/java/co/elastic/apm/agent/es/restclient/v5_6/ElasticsearchClientAsyncInstrumentation.java

```diff
@@ -77,32 +75,37 @@ public ElasticsearchClientAsyncInstrumentation(ElasticApmTracer tracer) {
 
 
     public static class ElasticsearchRestClientAsyncAdvice {
-        @VisibleForAdvice
-        public static final GlobalThreadLocal<Span> spanTls = GlobalThreadLocal.get(ElasticsearchRestClientAsyncAdvice.class, "spanTls");
-        @AssignTo.Argument(5)
+
+        @Nullable
+        @AssignTo.Argument(index = 1, value = 5)
         @Advice.OnMethodEnter(suppress = Throwable.class)
-        public static ResponseListener onBeforeExecute(@Advice.Argument(0) String method,
-                                                       @Advice.Argument(1) String endpoint,
-                                                       @Advice.Argument(3) @Nullable HttpEntity entity,
-                                                       @Advice.Argument(5) ResponseListener responseListener) {
+        public static Object[] onBeforeExecute(@Advice.Argument(0) String method,
+                                               @Advice.Argument(1) String endpoint,
+                                               @Advice.Argument(3) @Nullable HttpEntity entity,
+                                               @Advice.Argument(5) ResponseListener responseListener) {
 
             ElasticsearchRestClientInstrumentationHelper<HttpEntity, Response, ResponseListener> helper = esClientInstrHelperManager.getForClassLoaderOfClass(Response.class);
             if (helper != null) {
                 Span span = helper.createClientSpan(method, endpoint, entity);
-                spanTls.set(span);
                 if (span != null) {
-                    return helper.<ResponseListener>wrapResponseListener(responseListener, span);
+                    Object[] ret = new Object[2];
+                    ret[0] = span;
+                    ret[1] = helper.<ResponseListener>wrapResponseListener(responseListener, span);
+                    return ret;
```

Comment:
```suggestion
                    return new Object[]{span, helper.<ResponseListener>wrapResponseListener(responseListener, span)};
```

feel free to ignore if you like yours better